WebCore::MouseButton should be an enum class

aprotyas opened the pull request:
#### 865af240558dd997c13e159713ef3504d8f6e3c5
<pre>
WebCore::MouseButton should be an enum class
<a href="https://bugs.webkit.org/show_bug.cgi?id=258909">https://bugs.webkit.org/show_bug.cgi?id=258909</a>
rdar://111820855

Reviewed by Wenson Hsieh.

Converting `WebCore::MouseButton` to an enum class prevents us from
assigning seemingly magic numbers (-1/-2) to the `button` property of a
`MouseEvent`. In other words, it improves readability because we don&apos;t
have to go back and forth between the DOM API for the meaning of various
integral values.

This commit also changes our internal representation of the `button`
property from a `short` to an `int16_t`. This represents a
small conformance win since the latter  is better aligned with the Web
IDL standard&apos;s definition of the `short` type.

To address undefined behavior concerns, we declare all expected
`MouseButton` cases together and only pass around `MouseButton` values
throughout the engine. This means that as long as we&apos;re safely
converting to/from integral button values received over IDL, we should
be behaving correctly.

I&apos;ve also taken the liberty to make the enumeration cases read less
redundantly: {LeftButton, RightButton, MiddleButton, NoButton} =&gt; {Left,
Right, Middle, None}.

* LayoutTests/fast/events/constructors/mouse-event-constructor.html:
* Source/WebCore/Modules/model-element/HTMLModelElement.cpp:
(WebCore::HTMLModelElement::defaultEventHandler):
* Source/WebCore/dom/DragEvent.cpp:
(WebCore::DragEvent::create):
(WebCore::DragEvent::DragEvent):
* Source/WebCore/dom/DragEvent.h:
* Source/WebCore/dom/Element.cpp:
(WebCore::Element::dispatchMouseForceWillBegin):
* Source/WebCore/dom/MouseEvent.cpp:
(WebCore::MouseEvent::create):
(WebCore::MouseEvent::MouseEvent):
(WebCore::MouseEvent::initMouseEvent):
(WebCore::MouseEvent::initMouseEventQuirk):
(WebCore::MouseEvent::canTriggerActivationBehavior):
(WebCore::MouseEvent::button const):
* Source/WebCore/dom/MouseEvent.h:
(WebCore::MouseEvent::buttonAsShort const):
(WebCore::MouseEvent::button const): Deleted.
* Source/WebCore/dom/MouseEvent.idl:
* Source/WebCore/dom/MouseEventInit.h:
* Source/WebCore/dom/Node.cpp:
(WebCore::Node::defaultEventHandler):
* Source/WebCore/dom/PointerEvent.cpp:
(WebCore::PointerEvent::create):
(WebCore::PointerEvent::PointerEvent):
* Source/WebCore/dom/PointerEvent.h:
* Source/WebCore/dom/SimulatedClick.cpp:
* Source/WebCore/dom/WheelEvent.cpp:
(WebCore::WheelEvent::WheelEvent):
(WebCore::WheelEvent::initWebKitWheelEvent):
* Source/WebCore/dom/ios/MouseEventIOS.cpp:
(WebCore::MouseEvent::create):
* Source/WebCore/dom/ios/PointerEventIOS.cpp:
(WebCore::buttonForType):
* Source/WebCore/html/HTMLAnchorElement.cpp:
(WebCore::HTMLAnchorElement::defaultEventHandler):
* Source/WebCore/html/HTMLInputElement.cpp:
(WebCore::HTMLInputElement::willDispatchEvent):
(WebCore::HTMLInputElement::defaultEventHandler):
* Source/WebCore/html/HTMLSelectElement.cpp:
(WebCore::HTMLSelectElement::menuListDefaultEventHandler):
(WebCore::HTMLSelectElement::listBoxDefaultEventHandler):
* Source/WebCore/html/RangeInputType.cpp:
(WebCore::RangeInputType::handleMouseDownEvent):
* Source/WebCore/html/shadow/SliderThumbElement.cpp:
(WebCore::SliderThumbElement::defaultEventHandler):
* Source/WebCore/html/shadow/SpinButtonElement.cpp:
(WebCore::SpinButtonElement::defaultEventHandler):
* Source/WebCore/html/shadow/TextControlInnerElements.cpp:
(WebCore::SearchFieldResultsButtonElement::defaultEventHandler):
(WebCore::SearchFieldCancelButtonElement::defaultEventHandler):
* Source/WebCore/loader/NavigationAction.h:
* Source/WebCore/page/AutoscrollController.cpp:
(WebCore::AutoscrollController::handleMouseReleaseEvent):
* Source/WebCore/page/ContextMenuController.cpp:
(WebCore::ContextMenuController::showContextMenuAt):
* Source/WebCore/page/DebugPageOverlays.cpp:
(WebCore::InteractionRegionOverlay::mouseEvent):
* Source/WebCore/page/DragController.cpp:
(WebCore::createMouseEvent):
* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::handleMousePressEventDoubleClick):
(WebCore::EventHandler::handleMousePressEventTripleClick):
(WebCore::EventHandler::handleMousePressEventSingleClick):
(WebCore::EventHandler::handleMouseDraggedEvent):
(WebCore::EventHandler::eventMayStartDrag const):
(WebCore::EventHandler::handleMouseReleaseEvent):
(WebCore::EventHandler::handleMouseDoubleClickEvent):
(WebCore::EventHandler::dispatchDragEvent):
(WebCore::EventHandler::sendContextMenuEventForKey):
(WebCore::EventHandler::fakeMouseMoveEventTimerFired):
(WebCore::EventHandler::handleDrag):
* Source/WebCore/page/PointerCaptureController.cpp:
(WebCore::PointerCaptureController::pointerEventForMouseEvent):
* Source/WebCore/page/PointerCaptureController.h:
* Source/WebCore/page/ResourceUsageOverlay.cpp:
(WebCore::ResourceUsageOverlay::mouseEvent):
* Source/WebCore/page/ios/EventHandlerIOS.mm:
(WebCore::EventHandler::tryToBeginDragAtPoint):
* Source/WebCore/page/mac/ServicesOverlayController.mm:
(WebCore::ServicesOverlayController::mouseEvent):
* Source/WebCore/platform/PlatformMouseEvent.h:
* Source/WebCore/platform/ScrollbarTheme.cpp:
(WebCore::ScrollbarTheme::handleMousePressEvent):
* Source/WebCore/platform/adwaita/ScrollbarThemeAdwaita.cpp:
(WebCore::ScrollbarThemeAdwaita::handleMousePressEvent):
* Source/WebCore/platform/gtk/ScrollbarThemeGtk.cpp:
(WebCore::ScrollbarThemeGtk::handleMousePressEvent):
* Source/WebCore/platform/ios/PlatformEventFactoryIOS.mm:
(WebCore::PlatformMouseEventBuilder::PlatformMouseEventBuilder):
* Source/WebCore/platform/mac/PlatformEventFactoryMac.mm:
(WebCore::mouseButtonForEvent):
* Source/WebCore/platform/mac/ScrollbarThemeMac.mm:
(WebCore::shouldCenterOnThumb):
(WebCore::ScrollbarThemeMac::handleMousePressEvent):
* Source/WebCore/platform/win/PlatformMouseEventWin.cpp:
(WebCore::PlatformMouseEvent::PlatformMouseEvent):
* Source/WebCore/rendering/RenderEmbeddedObject.cpp:
(WebCore::RenderEmbeddedObject::handleUnavailablePluginIndicatorEvent):
* Source/WebCore/rendering/RenderFrameSet.cpp:
(WebCore::RenderFrameSet::userResize):
* Source/WebKit/Shared/API/c/WKSharedAPICast.h:
(WebKit::toAPI):
* Source/WebKit/Shared/NavigationActionData.h:
* Source/WebKit/Shared/WebEvent.serialization.in:
* Source/WebKit/Shared/WebEventConversion.cpp:
(WebKit::WebKit2PlatformMouseEvent::WebKit2PlatformMouseEvent):
* Source/WebKit/Shared/WebMouseEvent.cpp:
(WebKit::mouseButton):
* Source/WebKit/Shared/WebMouseEvent.h:
(WebKit::WebMouseEvent::button const):
* Source/WebKit/Shared/gtk/WebEventFactory.cpp:
(WebKit::buttonForEvent):
(WebKit::WebEventFactory::createWebMouseEvent):
* Source/WebKit/Shared/ios/WebIOSEventFactory.mm:
(WebIOSEventFactory::createWebMouseEvent):
* Source/WebKit/Shared/libwpe/WebEventFactory.cpp:
(WebKit::WebEventFactory::createWebMouseEvent):
* Source/WebKit/Shared/mac/WebEventFactory.mm:
(WebKit::currentMouseButton):
(WebKit::mouseButtonForEvent):
(WebKit::WebEventFactory::toNSButtonNumber):
* Source/WebKit/Shared/win/WebEventFactory.cpp:
(WebKit::clickCount):
(WebKit::WebEventFactory::createWebMouseEvent):
* Source/WebKit/UIProcess/API/glib/WebKitPrivate.cpp:
(toWebKitMouseButton):
* Source/WebKit/UIProcess/API/gtk/WebKitWebViewBase.cpp:
(MotionEvent::MotionEvent):
(webkitWebViewBaseSynthesizeMouseEvent):
* Source/WebKit/UIProcess/Automation/mac/WebAutomationSessionMac.mm:
(WebKit::automationMouseButtonToPlatformMouseButton):
(WebKit::WebAutomationSession::platformSimulateMouseInteraction):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::handleMouseEvent):
(WebKit::WebPageProxy::processNextQueuedMouseEvent):
(WebKit::WebPageProxy::didReceiveEvent):
* Source/WebKit/UIProcess/gtk/PointerLockManager.h:
* Source/WebKit/UIProcess/ios/WKMouseInteraction.mm:
(-[WKMouseInteraction createMouseEventWithType:wasCancelled:]):
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePageOverlay.cpp:
* Source/WebKit/WebProcess/InjectedBundle/API/c/mac/WKBundlePageBannerMac.mm:
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMMouseEvent.cpp:
(webkit_dom_mouse_event_init_mouse_event):
(webkit_dom_mouse_event_get_button):
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(WebKit::getEventTypeFromWebEvent):
(WebKit::PDFPlugin::handleMouseEvent):
(WebKit::PDFPlugin::showContextMenuAtPoint):
* Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.cpp:
(WebKit::WebLocalFrameLoaderClient::didSameDocumentNavigationForFrameViaJSHistoryAPI):
* Source/WebKit/WebProcess/WebPage/WebFrame.cpp:
(WebKit::isContextClick):
(WebKit::WebFrame::handleMouseEvent):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::navigateToPDFLinkWithSimulatedClick):
(WebKit::WebPage::contextMenuAtPointInWindow):
(WebKit::WebPage::mouseEvent):
(WebKit::WebPage::dragEnded):
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::dispatchSyntheticMouseMove):
(WebKit::WebPage::completeSyntheticClick):
(WebKit::WebPage::handleDoubleTapForDoubleClickAtPoint):
(WebKit::WebPage::requestAdditionalItemsForDragSession):
(WebKit::WebPage::inspectorNodeSearchMovedToPosition):
(WebKit::WebPage::dispatchSyntheticMouseEventsForSelectionGesture):
* Source/WebKitLegacy/ios/WebView/WebPDFViewPlaceholder.mm:
(-[WebPDFViewPlaceholder simulateClickOnLinkToURL:]):
* Source/WebKitLegacy/mac/DOM/DOMMouseEvent.mm:
(-[DOMMouseEvent button]):
* Source/WebKitLegacy/mac/WebCoreSupport/WebFrameLoaderClient.mm:
(WebFrameLoaderClient::actionDictionary const):
* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
(-[WebFrame _dragSourceEndedAt:operation:]):
* Source/WebKitLegacy/mac/WebView/WebPDFView.mm:
(-[WebPDFView PDFViewWillClickOnLink:withURL:]):
* Tools/TestWebKitAPI/Tests/mac/MenuTypesForMouseEvents.mm:
(TestWebKitAPI::TEST):

Canonical link: <a href="https://commits.webkit.org/268053@main">https://commits.webkit.org/268053@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b78e09b800bc3ca0976505b157d6ecc972f359c7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/18527 "4 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/18861 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/19466 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/20383 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/17329 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/18722 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/22174 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/19006 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/19205 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/18751 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/18926 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/16135 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/21259 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/16148 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/16897 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/23352 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/17167 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/17067 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/21247 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/17646 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/14956 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/16706 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/16729 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4404 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/21070 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/17488 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas reviewed (commented):
`style-checker` is failing locally, but these are all false positives and/or existing issues.

```
ERROR: Source/WebCore/dom/MouseEvent.cpp:86:  Wrong number of spaces before statement. (expected: 8)  [whitespace/indent] [4]
ERROR: Source/WebCore/dom/MouseEvent.cpp:88:  Wrong number of spaces before statement. (expected: 8)  [whitespace/indent] [4]
ERROR: Source/WebCore/dom/MouseEvent.cpp:89:  Wrong number of spaces before statement. (expected: 8)  [whitespace/indent] [4]
ERROR: Source/WebCore/platform/PlatformMouseEvent.h:42:  Code inside a namespace should not be indented.  [whitespace/indent] [4]
Total errors found: 4 in 51 files
```

rr-codes commented on Source/WebCore/platform/PlatformMouseEvent.h:
```diff
@@ -39,7 +39,7 @@ const double ForceAtForceClick = 2;
     // We use -2 for NoButton because -1 is a valid value in the DOM API for Pointer Events for pointermove events that
     // indicate that the pressed mouse button hasn't changed since the last event.
-    enum MouseButton : int8_t { LeftButton = 0, MiddleButton, RightButton, NoButton = -2 };
+    enum class MouseButton : int16_t { LeftButton = 0, MiddleButton, RightButton, NoButton = -2 };
```
these should probably be `Left, Middle, Right, None` I think

rr-codes commented on Source/WebKitLegacy/mac/WebCoreSupport/WebFrameLoaderClient.mm:
```diff
@@ -1553,10 +1553,11 @@ static void addRedirectURL(WebHistoryItem *item, const String& url)
+        const auto toUnderlyingType = [](MouseButton button) {
+            return static_cast<std::underlying_type_t<MouseButton>>(button);
+        };
+        auto button = mouseEventData->isTrusted ? toUnderlyingType(mouseEventData->button) : toUnderlyingType(MouseButton::NoButton);
```
we already have `WTF::enumToUnderlyingType` that you can use I think

aprotyas replied:
Oh, TIL. I'll change it

whsieh commented on Source/WebKit/Shared/WebMouseEvent.h:
```diff
@@ -43,7 +43,8 @@ namespace WebKit {
 enum class GestureWasCancelled : bool { No, Yes };
 
-enum class WebMouseEventButton : int32_t {
+// FIXME [aprotyas]: make this an enum class with int8_t backing type as well
```
Nit - we generally do `FIXME (Bugzilla number): …` instead of `FIXME (username): …` in WebKit.

aprotyas replied:
Yikes good catch, this comment is (a) wrong and (b) outdated... not sure how I committed it

whsieh commented on Source/WebKitLegacy/mac/WebCoreSupport/WebFrameLoaderClient.mm:
```diff
@@ -1553,10 +1553,11 @@ static void addRedirectURL(WebHistoryItem *item, const String& url)
+        const auto toUnderlyingType = [](MouseButton button) {
+            return static_cast<std::underlying_type_t<MouseButton>>(button);
+        };
+        auto button = mouseEventData->isTrusted ? toUnderlyingType(mouseEventData->button) : toUnderlyingType(MouseButton::NoButton);
```
This might actually be simpler as:

```suggestion
        auto button = static_cast<std::underlying_type_t<MouseButton>>(mouseEventData->isTrusted ? mouseEventData->button : MouseButton::NoButton);
```

(and then you don't need the lambda)

aprotyas replied:
Fixed in https://github.com/WebKit/WebKit/commit/4922291bd8adbd2dfa8a9054b5ac748a657ca4da

whsieh commented on Source/WebCore/dom/MouseEvent.cpp:
```diff
@@ -80,36 +80,36 @@ MouseEvent::MouseEvent() = default;
 MouseEvent::MouseEvent(const AtomString& eventType, const MouseEventInit& initializer)
     : MouseRelatedEvent(eventType, initializer)
-    , m_button(initializer.button == -2 ? 0 : initializer.button)
+    , m_button(static_cast<MouseButton>(initializer.button) == MouseButton::NoButton ? MouseButton::LeftButton : static_cast<MouseButton>(initializer.button))
```
Hm... I think this is probably fine, but it might be worth double checking with some of our security folks (i.e. whether this constitutes UB).

whsieh replied:
(It seems like we should _probably_ avoid this after all)

aprotyas replied:
Fixed in https://github.com/WebKit/WebKit/commit/dac58bfd6a4a82417491d6272fb7f66ba081b565

aprotyas committed:
WebCore::MouseButton should be an enum class
https://bugs.webkit.org/show_bug.cgi?id=258909
rdar://111820855

Reviewed by Wenson Hsieh.

Converting `WebCore::MouseButton` to an enum class prevents us from
assigning seemingly magic numbers (-1/-2) to the `button` property of a
`MouseEvent`. In other words, it improves readability because we don't
have to go back and forth between the DOM API for the meaning of various
integral values.

This commit also changes our internal representation of the `button`
property from a `short` to an `int16_t`. This represents a
small conformance win since the latter  is better aligned with the Web
IDL standard's definition of the `short` type.

To address undefined behavior concerns, we declare all expected
`MouseButton` cases together and only pass around `MouseButton` values
throughout the engine. This means that as long as we're safely
converting to/from integral button values received over IDL, we should
be behaving correctly.

I've also taken the liberty to make the enumeration cases read less
redundantly: {LeftButton, RightButton, MiddleButton, NoButton} => {Left,
Right, Middle, None}.

* LayoutTests/fast/events/constructors/mouse-event-constructor.html:
* Source/WebCore/Modules/model-element/HTMLModelElement.cpp:
(WebCore::HTMLModelElement::defaultEventHandler):
* Source/WebCore/dom/DragEvent.cpp:
(WebCore::DragEvent::create):
(WebCore::DragEvent::DragEvent):
* Source/WebCore/dom/DragEvent.h:
* Source/WebCore/dom/Element.cpp:
(WebCore::Element::dispatchMouseForceWillBegin):
* Source/WebCore/dom/MouseEvent.cpp:
(WebCore::MouseEvent::create):
(WebCore::MouseEvent::MouseEvent):
(WebCore::MouseEvent::initMouseEvent):
(WebCore::MouseEvent::initMouseEventQuirk):
(WebCore::MouseEvent::canTriggerActivationBehavior):
(WebCore::MouseEvent::button const):
* Source/WebCore/dom/MouseEvent.h:
(WebCore::MouseEvent::buttonAsShort const):
(WebCore::MouseEvent::button const): Deleted.
* Source/WebCore/dom/MouseEvent.idl:
* Source/WebCore/dom/MouseEventInit.h:
* Source/WebCore/dom/Node.cpp:
(WebCore::Node::defaultEventHandler):
* Source/WebCore/dom/PointerEvent.cpp:
(WebCore::PointerEvent::create):
(WebCore::PointerEvent::PointerEvent):
* Source/WebCore/dom/PointerEvent.h:
* Source/WebCore/dom/SimulatedClick.cpp:
* Source/WebCore/dom/WheelEvent.cpp:
(WebCore::WheelEvent::WheelEvent):
(WebCore::WheelEvent::initWebKitWheelEvent):
* Source/WebCore/dom/ios/MouseEventIOS.cpp:
(WebCore::MouseEvent::create):
* Source/WebCore/dom/ios/PointerEventIOS.cpp:
(WebCore::buttonForType):
* Source/WebCore/html/HTMLAnchorElement.cpp:
(WebCore::HTMLAnchorElement::defaultEventHandler):
* Source/WebCore/html/HTMLInputElement.cpp:
(WebCore::HTMLInputElement::willDispatchEvent):
(WebCore::HTMLInputElement::defaultEventHandler):
* Source/WebCore/html/HTMLSelectElement.cpp:
(WebCore::HTMLSelectElement::menuListDefaultEventHandler):
(WebCore::HTMLSelectElement::listBoxDefaultEventHandler):
* Source/WebCore/html/RangeInputType.cpp:
(WebCore::RangeInputType::handleMouseDownEvent):
* Source/WebCore/html/shadow/SliderThumbElement.cpp:
(WebCore::SliderThumbElement::defaultEventHandler):
* Source/WebCore/html/shadow/SpinButtonElement.cpp:
(WebCore::SpinButtonElement::defaultEventHandler):
* Source/WebCore/html/shadow/TextControlInnerElements.cpp:
(WebCore::SearchFieldResultsButtonElement::defaultEventHandler):
(WebCore::SearchFieldCancelButtonElement::defaultEventHandler):
* Source/WebCore/loader/NavigationAction.h:
* Source/WebCore/page/AutoscrollController.cpp:
(WebCore::AutoscrollController::handleMouseReleaseEvent):
* Source/WebCore/page/ContextMenuController.cpp:
(WebCore::ContextMenuController::showContextMenuAt):
* Source/WebCore/page/DebugPageOverlays.cpp:
(WebCore::InteractionRegionOverlay::mouseEvent):
* Source/WebCore/page/DragController.cpp:
(WebCore::createMouseEvent):
* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::handleMousePressEventDoubleClick):
(WebCore::EventHandler::handleMousePressEventTripleClick):
(WebCore::EventHandler::handleMousePressEventSingleClick):
(WebCore::EventHandler::handleMouseDraggedEvent):
(WebCore::EventHandler::eventMayStartDrag const):
(WebCore::EventHandler::handleMouseReleaseEvent):
(WebCore::EventHandler::handleMouseDoubleClickEvent):
(WebCore::EventHandler::dispatchDragEvent):
(WebCore::EventHandler::sendContextMenuEventForKey):
(WebCore::EventHandler::fakeMouseMoveEventTimerFired):
(WebCore::EventHandler::handleDrag):
* Source/WebCore/page/PointerCaptureController.cpp:
(WebCore::PointerCaptureController::pointerEventForMouseEvent):
* Source/WebCore/page/PointerCaptureController.h:
* Source/WebCore/page/ResourceUsageOverlay.cpp:
(WebCore::ResourceUsageOverlay::mouseEvent):
* Source/WebCore/page/ios/EventHandlerIOS.mm:
(WebCore::EventHandler::tryToBeginDragAtPoint):
* Source/WebCore/page/mac/ServicesOverlayController.mm:
(WebCore::ServicesOverlayController::mouseEvent):
* Source/WebCore/platform/PlatformMouseEvent.h:
* Source/WebCore/platform/ScrollbarTheme.cpp:
(WebCore::ScrollbarTheme::handleMousePressEvent):
* Source/WebCore/platform/adwaita/ScrollbarThemeAdwaita.cpp:
(WebCore::ScrollbarThemeAdwaita::handleMousePressEvent):
* Source/WebCore/platform/gtk/ScrollbarThemeGtk.cpp:
(WebCore::ScrollbarThemeGtk::handleMousePressEvent):
* Source/WebCore/platform/ios/PlatformEventFactoryIOS.mm:
(WebCore::PlatformMouseEventBuilder::PlatformMouseEventBuilder):
* Source/WebCore/platform/mac/PlatformEventFactoryMac.mm:
(WebCore::mouseButtonForEvent):
* Source/WebCore/platform/mac/ScrollbarThemeMac.mm:
(WebCore::shouldCenterOnThumb):
(WebCore::ScrollbarThemeMac::handleMousePressEvent):
* Source/WebCore/platform/win/PlatformMouseEventWin.cpp:
(WebCore::PlatformMouseEvent::PlatformMouseEvent):
* Source/WebCore/rendering/RenderEmbeddedObject.cpp:
(WebCore::RenderEmbeddedObject::handleUnavailablePluginIndicatorEvent):
* Source/WebCore/rendering/RenderFrameSet.cpp:
(WebCore::RenderFrameSet::userResize):
* Source/WebKit/Shared/API/c/WKSharedAPICast.h:
(WebKit::toAPI):
* Source/WebKit/Shared/NavigationActionData.h:
* Source/WebKit/Shared/WebEvent.serialization.in:
* Source/WebKit/Shared/WebEventConversion.cpp:
(WebKit::WebKit2PlatformMouseEvent::WebKit2PlatformMouseEvent):
* Source/WebKit/Shared/WebMouseEvent.cpp:
(WebKit::mouseButton):
* Source/WebKit/Shared/WebMouseEvent.h:
(WebKit::WebMouseEvent::button const):
* Source/WebKit/Shared/gtk/WebEventFactory.cpp:
(WebKit::buttonForEvent):
(WebKit::WebEventFactory::createWebMouseEvent):
* Source/WebKit/Shared/ios/WebIOSEventFactory.mm:
(WebIOSEventFactory::createWebMouseEvent):
* Source/WebKit/Shared/libwpe/WebEventFactory.cpp:
(WebKit::WebEventFactory::createWebMouseEvent):
* Source/WebKit/Shared/mac/WebEventFactory.mm:
(WebKit::currentMouseButton):
(WebKit::mouseButtonForEvent):
(WebKit::WebEventFactory::toNSButtonNumber):
* Source/WebKit/Shared/win/WebEventFactory.cpp:
(WebKit::clickCount):
(WebKit::WebEventFactory::createWebMouseEvent):
* Source/WebKit/UIProcess/API/glib/WebKitPrivate.cpp:
(toWebKitMouseButton):
* Source/WebKit/UIProcess/API/gtk/WebKitWebViewBase.cpp:
(MotionEvent::MotionEvent):
(webkitWebViewBaseSynthesizeMouseEvent):
* Source/WebKit/UIProcess/Automation/mac/WebAutomationSessionMac.mm:
(WebKit::automationMouseButtonToPlatformMouseButton):
(WebKit::WebAutomationSession::platformSimulateMouseInteraction):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::handleMouseEvent):
(WebKit::WebPageProxy::processNextQueuedMouseEvent):
(WebKit::WebPageProxy::didReceiveEvent):
* Source/WebKit/UIProcess/gtk/PointerLockManager.h:
* Source/WebKit/UIProcess/ios/WKMouseInteraction.mm:
(-[WKMouseInteraction createMouseEventWithType:wasCancelled:]):
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePageOverlay.cpp:
* Source/WebKit/WebProcess/InjectedBundle/API/c/mac/WKBundlePageBannerMac.mm:
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMMouseEvent.cpp:
(webkit_dom_mouse_event_init_mouse_event):
(webkit_dom_mouse_event_get_button):
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(WebKit::getEventTypeFromWebEvent):
(WebKit::PDFPlugin::handleMouseEvent):
(WebKit::PDFPlugin::showContextMenuAtPoint):
* Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.cpp:
(WebKit::WebLocalFrameLoaderClient::didSameDocumentNavigationForFrameViaJSHistoryAPI):
* Source/WebKit/WebProcess/WebPage/WebFrame.cpp:
(WebKit::isContextClick):
(WebKit::WebFrame::handleMouseEvent):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::navigateToPDFLinkWithSimulatedClick):
(WebKit::WebPage::contextMenuAtPointInWindow):
(WebKit::WebPage::mouseEvent):
(WebKit::WebPage::dragEnded):
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::dispatchSyntheticMouseMove):
(WebKit::WebPage::completeSyntheticClick):
(WebKit::WebPage::handleDoubleTapForDoubleClickAtPoint):
(WebKit::WebPage::requestAdditionalItemsForDragSession):
(WebKit::WebPage::inspectorNodeSearchMovedToPosition):
(WebKit::WebPage::dispatchSyntheticMouseEventsForSelectionGesture):
* Source/WebKitLegacy/ios/WebView/WebPDFViewPlaceholder.mm:
(-[WebPDFViewPlaceholder simulateClickOnLinkToURL:]):
* Source/WebKitLegacy/mac/DOM/DOMMouseEvent.mm:
(-[DOMMouseEvent button]):
* Source/WebKitLegacy/mac/WebCoreSupport/WebFrameLoaderClient.mm:
(WebFrameLoaderClient::actionDictionary const):
* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
(-[WebFrame _dragSourceEndedAt:operation:]):
* Source/WebKitLegacy/mac/WebView/WebPDFView.mm:
(-[WebPDFView PDFViewWillClickOnLink:withURL:]):
* Tools/TestWebKitAPI/Tests/mac/MenuTypesForMouseEvents.mm:
(TestWebKitAPI::TEST):

Canonical link: https://commits.webkit.org/268053@main
